### PR TITLE
Add error handling section

### PR DESCRIPTION
**Pull request by PatStLouis**
Addresses #383 #340 #331 and #141 

Standing questions:

[RFC 9457](https://www.rfc-editor.org/rfc/rfc9457.html) or [RFC 7807](https://datatracker.ietf.org/doc/html/rfc7807)?

What should the relationship with [verification algorithms](https://w3c.github.io/vc-data-model/#algorithms) from the spec be?

Should we define specific error codes for each [ProblemDetails](https://w3c.github.io/vc-data-model/#problem-details)?

Is it fair to assume that if an errors are returned the credential is not verified but if only warning are returned it is verified? (see list of error/warnings in the PR for details)

Keyword [status](https://datatracker.ietf.org/doc/html/rfc7807#section-3.1) vs code...

***
<a href="https://pr-preview.s3.amazonaws.com/OpSecId/vc-api/pull/389.html" title="Last updated on Jun 18, 2024, 7:20 PM UTC (bc3b5c5)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c-ccg/vc-api/389/c02e73f...OpSecId:bc3b5c5.html" title="Last updated on Jun 18, 2024, 7:20 PM UTC (bc3b5c5)">Diff</a>

### CODE REVIEW

**Review by TallTed — CHANGES_REQUESTED** [review]

**Review by TallTed — APPROVED** [review]

**Comment by msporny** [comment]
The group discussed this on the 2024-06-04 telecon:

> RFC 9457 or RFC 7807?

RFC 9457 should be used.

> What should the relationship with verification algorithms from the spec be?

We shouldn't tie too closely to VCDM v2.0 since the API could be used to issue mDL, mDoc, and other types of digital credentials.

> Should we define specific error codes for each ProblemDetails?

Not yet. We need to make sure we don't mix "HTTP Error Codes" with "ProblemDetails" codes. We need to update the ProblemDetails object to align w/ RFC 9457 (add the "status" field). We need to also figure out (later, in another PR) what to do for things like "error revoked" when validating a VC, and so on.

> Is it fair to assume that if an errors are returned the credential is not verified but if only warning are returned it is verified? (see list of error/warnings in the PR for details)

This feels like a safe assumption to make.

> Keyword status vs code...

"status" is for "HTTP Error Codes" (200, 400, etc.) and "code" is for integer codes returned by low-level libraries.

File: index.html [hunk 1]
```diff
@@ -1207,27 +1207,37 @@ <h4>Relationship between verification and error handling</h4>
...
-          "warnings": [ProblemDetails],
-          "errors": [ProblemDetails]
-        }
-    </code>
+        <h5>Errors and Warnings</h5>
+        Errors are ProblemDetails relating to cryptography, data model and malformed context.
+        Warnings are ProblemDetails relating to statuses and validity periods.
+
+        The VerificationResponse object MUST have the verified value set to false if an error is included.
+        The VerificationResponse object MUST have the verified value set to true if no errors are included.
```
Comment (TallTed) [inline index.html:1227]:
```suggestion
        Errors are `ProblemDetails` relating to cryptography, data model and malformed context.
        Warnings are `ProblemDetails` relating to statuses and validity periods.

        The `VerificationResponse` object MUST have the verified value set to `false` if an error is included.
        The `VerificationResponse` object MUST have the verified value set to `true` if no errors are included.
```

**Review by TallTed — CHANGES_REQUESTED** [review]

**Review by aljones15 — APPROVED** [review]
LGTM

File: index.html [hunk 2]
```diff
@@ -1173,6 +1174,65 @@ <h4>Exchange Examples</h4>
...
+
+        <pre class="example"
+          title="Verification Response">
+          {
+            "verified": false,
+            "document": verifiableCredential,
+            "mediaType": "vc+json+ld",
```
Comment (msporny) [inline index.html:1227]:
```suggestion
            "mediaType": "application/vc",
```

**Pull request merged**